### PR TITLE
sublime-music: init at 0.11.0

### PR DESCRIPTION
**Pull request by albakham**
###### Motivation for this change

Package the sublime-music application.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/sublime-music/default.nix [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+   ++ lib.optional keyringSupport keyring
+   ++ lib.optional serverSupport bottle
+  ;
+
+  strictDeps = false;
+
+  meta = with lib; {
+    description = "Sublime Music is a GTK3 Subsonic/Airsonic/Revel/Gonic/Navidrome/*sonic client for the Linux Desktop.";
+    homepage = "https://sublimemusic.app/";
+    license = licenses.gpl3;
```
Comment (jtojnar) [inline pkgs/applications/audio/sublime-music/default.nix:55]: https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206

File: pkgs/applications/audio/sublime-music/default.nix [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    python-dateutil
+    requests
+    semver
+  ]
+   ++ lib.optional chromecastSupport PyChromecast
+   ++ lib.optional keyringSupport keyring
+   ++ lib.optional serverSupport bottle
+  ;
+
+  strictDeps = false;
```
Comment (jtojnar) [inline pkgs/applications/audio/sublime-music/default.nix:50]: Could you add a comment with explanation?

Comment (albakham) [reply pkgs/applications/audio/sublime-music/default.nix:50]: I'll add a reference to https://github.com/NixOS/nixpkgs/issues/56943

**Comment by albakham** [comment]
I splitted the commit to better understand the additions, and I did the modifications @jtojnar suggested.

File: pkgs/applications/audio/sublime-music/default.nix [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+   ++ lib.optional keyringSupport keyring
+   ++ lib.optional serverSupport bottle
+  ;
+
+  # hook for gobject-introspection doesn't like strictDeps
+  # https://github.com/NixOS/nixpkgs/issues/56943
+  strictDeps = false;
+
+  meta = with lib; {
+    description = "Sublime Music is a GTK3 Subsonic/Airsonic/Revel/Gonic/Navidrome/*sonic client for the Linux Desktop.";
```
Comment (jtojnar) [inline pkgs/applications/audio/sublime-music/default.nix:55]: Looks good, just one last nit: https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes

File: pkgs/development/python-modules/dataclasses-json/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  };
+
+  propagatedBuildInputs = [
+    stringcase
+    typing-inspect
+    marshmallow-enum
+  ];
+
+  meta = with lib; {
+    description = "This library provides a simple API for encoding and decoding dataclasses to and from JSON.";
```
Comment (jtojnar) [inline pkgs/development/python-modules/dataclasses-json/default.nix:25]: Same here.

**Review by jtojnar — APPROVED** [review]

**Comment by jtojnar** [comment]
Thanks.

**Pull request merged**